### PR TITLE
bc: add bc-gh as an alternative bc

### PR DESCRIPTION
**Pull request by gavinhoward**
This PR adds `bc-gh` as an alternative `bc` and `dc` package and adds an `alternatives=` line to the `bc` package.

This PR also fixes `xlint` on the `bc` package.

### CODE REVIEW

**Comment by bugcrazy** [comment]
The bc-gh is added default on Void Linux, and gnu-bc is deprecated as default.

File: srcpkgs/bc-gh/template [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'bc-gh'
+pkgname=bc-gh
+version=1.1.1
+revision=1
+wrksrc="bc-${version}"
+short_desc="Implementation of POSIX bc with GNU extensions"
+maintainer="maxice8 <thinkabit.ukim@gmail.com>"
```
Comment (maxice8) [inline srcpkgs/bc-gh/template]: Please use your name here

File: srcpkgs/bc/template [hunk 2]
```diff
@@ -6,12 +6,13 @@ build_style=gnu-configure
 configure_args="--with-readline"
 hostmakedepends="ed flex"
 makedepends="readline-devel"
-short_desc="An arbitrary precision numeric processing language"
+short_desc="Arbitrary precision numeric processing language"
 maintainer="Juan RP <xtraeme@voidlinux.org>"
-homepage="http://www.gnu.org/software/${pkgname}/"
 license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/bc/template]: please use the spdx identifier for bc, it is a gnu project so it most likely is GPL-3.0-or-later but please check

**Comment by maxice8** [comment]
@bugcrazy no, in the meantime gnu's bc will remain default

**Comment by maxice8** [comment]
please revbump bc otherwise the changes won't be picked up

File: srcpkgs/bc/template (hunk 2 above)
Comment (gavinhoward) [reply srcpkgs/bc/template]: @maxice8 I just checked. You are correct.

**Comment by gavinhoward** [comment]
All comments have been fixed, as far as I can tell.

**Comment by gavinhoward** [comment]
Looks like the GNU `bc` is the reason Travis CI failed...

File: srcpkgs/bc-gh/template [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+version=1.1.1
+revision=1
+wrksrc="bc-${version}"
+short_desc="Implementation of POSIX bc with GNU extensions"
+maintainer="Gavin D. Howard <yzena.tech@gmail.com>"
+license="BSD-2-Clause"
+homepage="https://github.com/gavinhoward/bc"
+distfiles="${homepage}/releases/download/${version}/bc-${version}.tar.xz"
+checksum=7e8d427c37223aa7da2dd83ef8503236834ed6c0bd0b94fcba041699e4bc125d
+alternatives="bc:bc:/usr/bin/bc-gh dc:dc:/usr/bin/dc-gh"
```
Comment (maxice8) [inline srcpkgs/bc-gh/template]: please put one per-line, prefix every new line with 1 whitespace

Comment (gavinhoward) [reply srcpkgs/bc-gh/template]: Done.

File: srcpkgs/bc/template [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
-short_desc="An arbitrary precision numeric processing language"
+short_desc="Arbitrary precision numeric processing language"
 maintainer="Juan RP <xtraeme@voidlinux.org>"
+license="GPL-3.0-or-later"
 homepage="http://www.gnu.org/software/${pkgname}/"
-license="GPL-3"
 distfiles="${GNU_SITE}/${pkgname}/${pkgname}-${version}.tar.gz"
 checksum=62adfca89b0a1c0164c2cdca59ca210c1d44c3ffc46daf9931cf4942664cb02a
+alternatives="
+ bc:bc:/usr/bin/bc
```
Comment (maxice8) [inline srcpkgs/bc/template]: you created a bc alternatives that symlinks bc to... bc

**Comment by maxice8** [comment]
@gavinhoward you can add this to the `bc` template so that it actually works.

```
post_install() {
       mv ${DESTDIR}/usr/bin/{,gnu-}bc
       mv ${DESTDIR}/usr/bin/{,gnu-}dc
}
```

Also separate this Pull Request into 2 commits, one for adding gh-bc and one for updating `bc` to have alternatives

File: srcpkgs/bc/template (hunk 4 above)
Comment (gavinhoward) [reply srcpkgs/bc/template]: I guess I did not understand the docs. Oops.

**Comment by gavinhoward** [comment]
@maxice8 I believe I have fulfilled your requests.

**Comment by maxice8** [comment]
Please rename the commit that introduces bc-gh to

New package: bc-gh


to follow our commit naming scheme

**Comment by gavinhoward** [comment]
Done.

**Pull request merged**